python/smashbox/config.py

```diff
@@ -0,0 +1,99 @@
+import logging
+import os.path
+import pickle
+from pathlib import Path
+from typing import Any, Literal
+
+import yaml
+from pydantic import BaseSettings
+
+
+logger = None
+
+# this should probably be moved into a utilities module
+def get_logger(name: str = "config", level: int | None = None) -> logging.Logger:
+    global logger
```

Comment:
There is no need to cache the logger, Python does this internally. You can call `getLogger` as often as you want to, it will return the same object every time (at least when not using multithreading, of course). The method can be considered idempotent.